alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt

```diff
@@ -412,9 +412,19 @@ class TransportIndexMonitorAction @Inject constructor(
                     )
                     return
                 }
+                metadata = metadata.copy(monitorId = indexResponse.id)
+
+                val metadataIndexRequest = IndexRequest(SCHEDULED_JOBS_INDEX)
```

Comment:
During the monitor execution, we will have to recreate the metadata. This is being handled by the Document Level Monitor as its the only monitor type currently utilizing it.

Reply:
Would you mind making a comment about that here? That's a general error case that we seem to only be handling in Doc-Level Monitors. If we add its usage in other types, we'll need to remember to account for it. Ideally since this is being added to Monitors in general, we should add it to some base initialization logic so all Monitor types handle it by default (with the option to override the default values or something). We don't have to do that in this PR but the comment will be helpful in the meantime.

Reply:
I would highly insist on abstracting out the Metadata creation as a separate logic with clear interfaces and exceptions, Same can be re-used across if other monitors start using it. 
Let me know if that is something we can only do as a followup.